### ISSUE TITLE
Fix CI issues

### ISSUE THREAD

**Issue by david-allison-1** [david-allison-1/Anki-Android-Backend#37]
CI has started failing gradually - investigate why and fix it

https://github.com/david-allison-1/Anki-Android-Backend/actions

**Comment by mikehardy** [comment]
We're all just sliding inevitably towards the heat death of the universe, there's no escape

```
 entropy noun
Save Word

To save this word, you'll need to log in.
Log In 
en·​tro·​py | \ ˈen-trə-pē
\
plural entropies
Definition of entropy

1 thermodynamics : a measure of the unavailable energy in a closed thermodynamic system that is also usually considered to be a measure of the system's disorder, that is a property of the system's state, and that varies directly with any reversible change in heat in the system and inversely with the temperature of the system broadly : the degree of disorder or uncertainty in a system
2a : the degradation of the matter and energy in the universe to an ultimate state of inert uniformity Entropy is the general trend of the universe toward death and disorder.— James R. Newman
b : a process of degradation or running down or a trend to disorder The deterioration of copy editing and proof-reading, incidentally, is a token of the cultural entropy that has overtaken us in the postwar years.— John Simon

```

**Comment by david-allison-1** [comment]
<details><summary>Linux - Python (good)</summary>

```
Run sudo apt-get install python3-setuptools && pip3 install protobuf-compiler && .github/scripts/protoc_gen_deps.py
Reading package lists...
Building dependency tree...
Reading state information...
Suggested packages:
  python-setuptools-doc
The following NEW packages will be installed:
  python3-setuptools
0 upgraded, 1 newly installed, 0 to remove and 17 not upgraded.
Need to get 248 kB of archives.
After this operation, 1319 kB of additional disk space will be used.
Get:1 http://azure.archive.ubuntu.com/ubuntu bionic/main amd64 python3-setuptools all 39.0.1-2 [248 kB]
Fetched 248 kB in 0s (17.4 MB/s)
Selecting previously unselected package python3-setuptools.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... 90%
(Reading database ... 95%
(Reading database ... 100%
(Reading database ... 307282 files and directories currently installed.)
Preparing to unpack .../python3-setuptools_39.0.1-2_all.deb ...
Unpacking python3-setuptools (39.0.1-2) ...
Setting up python3-setuptools (39.0.1-2) ...
Collecting protobuf-compiler
  Downloading https://files.pythonhosted.org/packages/bb/d6/eef8da430f8a6df739d9de807db904a7b4c4103cb292f178201ca7e1121d/protobuf_compiler-1.0.20-py3-none-any.whl
Collecting bleach==2.1.0 (from protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/ab/92/ea5e4050d3a1fbc21d8e2ab6613175d90516bf8dad130fe14d2cac03f41b/bleach-2.1-py2.py3-none-any.whl
Collecting colorama==0.3.3 (from protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/24/84/29ce4167d1f5c4a320aaad91e1178e5a1baf9cfe1c63f9077c5dade0e3cc/colorama-0.3.3.tar.gz
Collecting grpcio-tools==1.18.0 (from protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/b9/9b/a31e21e49ae9732dfa42b3c60cf3d156dd0bb14bb73a5c718b24b4c6cefd/grpcio_tools-1.18.0-cp36-cp36m-manylinux1_x86_64.whl (22.8MB)
Collecting tqdm==4.31.1 (from protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/6c/4b/c38b5144cf167c4f52288517436ccafefe9dc01b8d1c190e18a6b154cd4a/tqdm-4.31.1-py2.py3-none-any.whl (48kB)
Collecting termcolor==1.1.0 (from protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/8a/48/a76be51647d0eb9f10e2a4511bf3ffb8cc1e6b14e9e4fab46173aa79f981/termcolor-1.1.0.tar.gz
Collecting grpcio==1.18.0 (from protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/51/a3/489ce6a67047263e9b0da8784b2925c4f89b688a6e33073c5bb6c4c2867f/grpcio-1.18.0-cp36-cp36m-manylinux1_x86_64.whl (10.6MB)
Collecting html5lib!=1.0b1,!=1.0b2,!=1.0b3,!=1.0b4,!=1.0b5,!=1.0b6,!=1.0b7,!=1.0b8,>=0.99999999pre (from bleach==2.1.0->protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/6c/dd/a834df6482147d48e225a49515aabc28974ad5a4ca3215c18a882565b028/html5lib-1.1-py2.py3-none-any.whl (112kB)
Collecting six (from bleach==2.1.0->protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/ee/ff/48bde5c0f013094d729fe4b0316ba2a24774b3ff1c52d924a8a4cb04078a/six-1.15.0-py2.py3-none-any.whl
Collecting protobuf>=3.5.0.post1 (from grpcio-tools==1.18.0->protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/fe/b9/1f775f8e304f6a0f0db313588d8842f5eee52d7dea9c3335fb4b4c7dd384/protobuf-3.15.5-cp36-cp36m-manylinux1_x86_64.whl (1.0MB)
Collecting webencodings (from html5lib!=1.0b1,!=1.0b2,!=1.0b3,!=1.0b4,!=1.0b5,!=1.0b6,!=1.0b7,!=1.0b8,>=0.99999999pre->bleach==2.1.0->protobuf-compiler)
  Downloading https://files.pythonhosted.org/packages/f4/24/2a3e3df732393fed8b3ebf2ec078f05546de641fe1b667ee316ec1dcf3b7/webencodings-0.5.1-py2.py3-none-any.whl
Building wheels for collected packages: colorama, termcolor
  Running setup.py bdist_wheel for colorama: started
  Running setup.py bdist_wheel for colorama: finished with status 'error'
  Failed building wheel for colorama
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-0_pafel4/colorama/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpts69khkypip-wheel- --python-tag cp36:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Running setup.py clean for colorama
  Running setup.py bdist_wheel for termcolor: started
  Running setup.py bdist_wheel for termcolor: finished with status 'error'
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-0_pafel4/termcolor/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpo6mj7ma0pip-wheel- --python-tag cp36:
  Failed building wheel for termcolor
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Running setup.py clean for termcolor
Failed to build colorama termcolor
Installing collected packages: six, webencodings, html5lib, bleach, colorama, protobuf, grpcio, grpcio-tools, tqdm, termcolor, protobuf-compiler
  Running setup.py install for colorama: started
    Running setup.py install for colorama: finished with status 'done'
  Running setup.py install for termcolor: started
    Running setup.py install for termcolor: finished with status 'done'
Successfully installed bleach-2.1 colorama-0.3.3 grpcio-1.18.0 grpcio-tools-1.18.0 html5lib-1.1 protobuf-3.15.5 protobuf-compiler-1.0.20 six-1.15.0 termcolor-1.1.0 tqdm-4.31.1 webencodings-0.5.1
Testing completed successfully
```


</details>

<details><summary>Linux - Python (bad)</summary>

```
Run sudo apt-get install python3-setuptools && pip3 install protobuf-compiler && .github/scripts/protoc_gen_deps.py
Reading package lists...
Building dependency tree...
Reading state information...
python3-setuptools is already the newest version (45.2.0-1).
0 upgraded, 0 newly installed, 0 to remove and 4 not upgraded.
Collecting protobuf-compiler
  Downloading protobuf_compiler-1.0.20-py3-none-any.whl (8.6 kB)
Collecting colorama==0.3.3
  Downloading colorama-0.3.3.tar.gz (22 kB)
Collecting grpcio-tools==1.18.0
  Downloading grpcio-tools-1.18.0.tar.gz (1.9 MB)
Collecting bleach==2.1.0
  Downloading bleach-2.1-py2.py3-none-any.whl (27 kB)
Collecting termcolor==1.1.0
  Downloading termcolor-1.1.0.tar.gz (3.9 kB)
Collecting tqdm==4.31.1
  Downloading tqdm-4.31.1-py2.py3-none-any.whl (48 kB)
Collecting grpcio==1.18.0
  Downloading grpcio-1.18.0.tar.gz (14.6 MB)
Requirement already satisfied: protobuf>=3.5.0.post1 in /usr/lib/python3/dist-packages (from grpcio-tools==1.18.0->protobuf-compiler) (3.6.1)
Collecting html5lib!=1.0b1,!=1.0b2,!=1.0b3,!=1.0b4,!=1.0b5,!=1.0b6,!=1.0b7,!=1.0b8,>=0.99999999pre
  Downloading html5lib-1.1-py2.py3-none-any.whl (112 kB)
Requirement already satisfied: six in /usr/lib/python3/dist-packages (from bleach==2.1.0->protobuf-compiler) (1.14.0)
Collecting webencodings
  Downloading webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Building wheels for collected packages: colorama, grpcio-tools, termcolor, grpcio
  Building wheel for colorama (setup.py): started
  Building wheel for colorama (setup.py): finished with status 'done'
  Created wheel for colorama: filename=colorama-0.3.3-py3-none-any.whl size=14319 sha256=1600da5f0c70b04bb5f3b523719608e23f869ac0388870041d6ddab3a9fc612e
  Stored in directory: /home/runner/.cache/pip/wheels/4a/76/18/7847dee1038bdd1122157607a54f730d12a5cdfdca2c0ceaff
  Building wheel for grpcio-tools (setup.py): started
  Building wheel for grpcio-tools (setup.py): still running...
  Building wheel for grpcio-tools (setup.py): still running...
  Building wheel for grpcio-tools (setup.py): still running...
  Building wheel for grpcio-tools (setup.py): still running...
  Building wheel for grpcio-tools (setup.py): still running...
  Building wheel for grpcio-tools (setup.py): still running...
  Building wheel for grpcio-tools (setup.py): finished with status 'done'
  Created wheel for grpcio-tools: filename=grpcio_tools-1.18.0-cp38-cp38-linux_x86_64.whl size=29418561 sha256=7b7b47a2a4967099f570eabea4624ff081e1c4f21a0cbed5d034a98ad4d681c5
  Stored in directory: /home/runner/.cache/pip/wheels/c3/a4/01/55498ffde60168ea7d2c237d95278e42bc7f0773bddee01581
  Building wheel for termcolor (setup.py): started
  Building wheel for termcolor (setup.py): finished with status 'done'
  Created wheel for termcolor: filename=termcolor-1.1.0-py3-none-any.whl size=4830 sha256=42d742dd23e08dbf2dccee62b6a467eb1a365a30ea7265ebea58b1417ec5fbf7
  Stored in directory: /home/runner/.cache/pip/wheels/a0/16/9c/5473df82468f958445479c59e784896fa24f4a5fc024b0f501
  Building wheel for grpcio (setup.py): started
  Building wheel for grpcio (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
      File "/usr/lib/python3.8/distutils/command/build_ext.py", line 449, in build_extensions
        self._build_extensions_serial()
      File "/usr/lib/python3.8/distutils/command/build_ext.py", line 474, in _build_extensions_serial
        self.build_extension(ext)
      File "/usr/lib/python3/dist-packages/setuptools/command/build_ext.py", line 208, in build_extension
        _build_ext.build_extension(self, ext)
      File "/usr/lib/python3.8/distutils/command/build_ext.py", line 528, in build_extension
        objects = self.compiler.compile(sources,
      File "/usr/lib/python3.8/distutils/ccompiler.py", line 574, in compile
        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
      File "/usr/lib/python3.8/distutils/unixccompiler.py", line 120, in _compile
        raise CompileError(msg)
    distutils.errors.CompileError: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-xbcynma_/grpcio/setup.py", line 353, in <module>
        setuptools.setup(
      File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 144, in setup
        return distutils.core.setup(**attrs)
      File "/usr/lib/python3.8/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/lib/python3.8/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/usr/lib/python3/dist-packages/setuptools/command/install.py", line 61, in run
        return orig.install.run(self)
      File "/usr/lib/python3.8/distutils/command/install.py", line 589, in run
        self.run_command('build')
      File "/usr/lib/python3.8/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/usr/lib/python3.8/distutils/command/build.py", line 135, in run
        self.run_command(cmd_name)
      File "/usr/lib/python3.8/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/usr/lib/python3/dist-packages/setuptools/command/build_ext.py", line 87, in run
        _build_ext.run(self)
      File "/usr/lib/python3.8/distutils/command/build_ext.py", line 340, in run
        self.build_extensions()
      File "/tmp/pip-install-xbcynma_/grpcio/src/python/grpcio/commands.py", line 276, in build_extensions
        raise CommandError(
    commands.CommandError: Failed `build_ext` step:
    Traceback (most recent call last):
      File "/usr/lib/python3.8/distutils/unixccompiler.py", line 117, in _compile
        self.spawn(compiler_so + cc_args + [src, '-o', obj] +
      File "/tmp/pip-install-xbcynma_/grpcio/src/python/grpcio/_spawn_patch.py", line 54, in _commandfile_spawn
        _classic_spawn(self, command)
      File "/usr/lib/python3.8/distutils/ccompiler.py", line 910, in spawn
        spawn(cmd, dry_run=self.dry_run)
      File "/usr/lib/python3.8/distutils/spawn.py", line 36, in spawn
        _spawn_posix(cmd, search_path, dry_run=dry_run)
      File "/usr/lib/python3.8/distutils/spawn.py", line 157, in _spawn_posix
        raise DistutilsExecError(
    distutils.errors.DistutilsExecError: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/tmp/pip-install-xbcynma_/grpcio/src/python/grpcio/commands.py", line 272, in build_extensions
        build_ext.build_ext.build_extensions(self)
      File "/usr/lib/python3.8/distutils/command/build_ext.py", line 449, in build_extensions
        self._build_extensions_serial()
      File "/usr/lib/python3.8/distutils/command/build_ext.py", line 474, in _build_extensions_serial
        self.build_extension(ext)
      File "/usr/lib/python3/dist-packages/setuptools/command/build_ext.py", line 208, in build_extension
        _build_ext.build_extension(self, ext)
      File "/usr/lib/python3.8/distutils/command/build_ext.py", line 528, in build_extension
        objects = self.compiler.compile(sources,
      File "/usr/lib/python3.8/distutils/ccompiler.py", line 574, in compile
        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
      File "/usr/lib/python3.8/distutils/unixccompiler.py", line 120, in _compile
        raise CompileError(msg)
    distutils.errors.CompileError: command 'x86_64-linux-gnu-gcc' failed with exit status 1
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-xbcynma_/grpcio/setup.py'"'"'; __file__='"'"'/tmp/pip-install-xbcynma_/grpcio/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-_2bl36nf/install-record.txt --single-version-externally-managed --user --prefix= --compile --install-headers /home/runner/.local/include/python3.8/grpcio Check the logs for full command output.
Error: Process completed with exit code 1.
```
</details>

**Comment by TarekkMA** [comment]
Python `protobuf-compiler` do cause issues on macOS as well, and that didn't get fixed until I downgraded to python 3.7.
The pip version have not been updated since Apr 10, 2019 and may depend on some outdated packages.

Is it possible to install it not from pip but from here? 
https://github.com/protocolbuffers/protobuf
https://grpc.io/docs/protoc-installation/

**Comment by david-allison-1** [comment]
Sorted, just running a check through CI now before I PR
